### ISSUE TITLE
Building .NET bindings for Z3 fails

### ISSUE THREAD

**Issue by suem** [mmjb/T2#22]
Hi,

I've been trying to build T2 as described in the build instructions on https://mmjb.github.io/T2

Unfortunately, I can't build the .NET bindings for Z3. The build fails with the following error message:

```
Build FAILED.
Errors:

/home/suem/workspace/z3-z3-4.5.0/src/api/dotnet/Microsoft.Z3.csproj (default targets) ->
/usr/lib/mono/xbuild/14.0/bin/Microsoft.CSharp.targets (CoreCompile target) ->

	CSC: error CS2001: Source file '/home/suem/workspace/z3-z3-4.5.0/src/api/dotnet/Enumerations.cs' could not be found.
	CSC: error CS2001: Source file '/home/suem/workspace/z3-z3-4.5.0/src/api/dotnet/Native.cs' could not be found.

	 0 Warning(s)
	 2 Error(s)
```

Building Z3 itself worked fine. But for some reason the 'Native.cs' and 'Enumerations.cs' files are missing.

My system is Arch Linux with the following version of mono:

`Mono JIT compiler version 5.0.0 (Stable 5.0.0.100/9667aa6 Fri May 12 07:20:13 UTC 2017)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           __thread
	SIGSEGV:       altstack
	Notifications: epoll
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug 
	LLVM:          supported, not enabled.
	GC:            sgen (concurrent by default)
`

I'm not that familiar with the .NET and mono platform. Any help would be greatly appreciated.

Cheers
Samuel

**Comment by mmjb** [comment]
Heya,

On Tue, Jun 6, 2017 at 12:51 PM, suem <notifications@github.com> wrote:
>
> I've been trying to build T2 as described in the build instructions on
> <https://mmjb.github.io/T2>
> ​​ <https://mmjb.github.io/T2>
> https://mmjb.github.io/T2
>
> Unfortunately, I can't build the .NET bindings for Z3. The build fails
> with the following error message:
>
> Build FAILED.
> Errors:
>
> /home/suem/workspace/z3-z3-4.5.0/src/api/dotnet/Microsoft.Z3.csproj (default targets) ->
> /usr/lib/mono/xbuild/14.0/bin/Microsoft.CSharp.targets (CoreCompile target) ->
>
> 	CSC: error CS2001: Source file '/home/suem/workspace/z3-z3-4.5.0/src/api/dotnet/Enumerations.cs' could not be found.
> 	CSC: error CS2001: Source file '/home/suem/workspace/z3-z3-4.5.0/src/api/dotnet/Native.cs' could not be found.
>
> 	 0 Warning(s)
> 	 2 Error(s)
>
> Building Z3 itself worked fine. But for some reason the 'Native.cs' and
> 'Enumerations.cs' files are missing.
>
> My system is Arch Linux with the following version of mono:
>
> Mono JIT compiler version 5.0.0 (Stable 5.0.0.100/9667aa6 Fri May 12
> 07:20:13 UTC 2017) Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and
> Contributors. www.mono-project.com TLS: __thread SIGSEGV: altstack
> Notifications: epoll Architecture: amd64 Disabled: none Misc: softdebug
> LLVM: supported, not enabled. GC: sgen (concurrent by default)
>
> I'm not that familiar with the .NET and mono platform. Any help would be
> greatly appreciated.
>

​Both Enumeration.cs and Native.cs should be auto-generated by the z3 build
process when you​ build the native part of it. This is now required --
however, I just noticed that
​
https://mmjb.github.io/T2 is outdated and does not state that. Please
follow the (slightly updated) instructions on
https://github.com/mmjb/T2/blob/master/README.txt (most notably, z3 4.5
won't do, and you instead need the spacer fork of it).

Marc

**Comment by suem** [comment]
Hi Marc,

Thanks a lot for your help. Sorry for taking so long to reply. 

I managed to build T2 on Ubuntu 16.04 with the updated instructions and some tinkering.

I created a [vagrant](https://www.vagrantup.com/) setup script that automatically builds T2 on a Virtualbox machine: [t2_vagrant.zip](https://github.com/mmjb/T2/files/1119433/t2_vagrant.zip)

Maybe this will prove useful to someone.

Cheers
Sam
